CONTRIBUTING.md

```diff
@@ -1,5 +1,84 @@
+
+## Questions
+
+For general problems you encounter that may not require a core maintainer to answer, post your question in our [community forum]("https://discuss.hashicorp.com/c/terraform-core/cdk-for-terraform/"). Otherwise if you believe the problem stems from a bug, please feel free to create a new issue describing the problem.
+
+## Issues
+
+This section goes over what we are looking for in regards to reported issues.
+
+### Reporting an Issue
+
+If you encounter a bug you can help us by submitting an issue, though make sure it is not covered by an existing Github issue. If you feel inclined to solve this issue yourself you can always submit a Pull Request with the fix– though be sure to create a separate issue (if one does not exist) containing the problem that can then be linked in your Pull Request.
+
+When reporting a bug we ask that you include:
+
+- CDKTF & language versions being used
+- Affected Resource(s)
+- Github Gist containing the Debug output
+- Expected Behavior vs. Actual Behavior
+- Steps to reproduce the issue
+- Other important info about the issue
+- Any relevant Github issues, pull requests, and/or documentation that provides greater context
+
+### Proposing a Feature
+
+In order to be respectful of the time of community contributors, we aim to discuss new potential feature and changes in GitHub issues prior to their implementation. This allows us to vet the potential utility of the change, as well as discuss how it could potentially fit into work currently being done, or a future milestone.
+
+When proposing a feature we ask that you include:
+
+- A description of the feature to add
+- Any relevant Github issues, pull requests, blog posts, and/or documentation that provides greater context
+
+## Pull Requests
+
+Your pull request should have a description of what it accomplishes. If applicable, PRs should include testing that validate correctness. In all cases the existing tests must pass.
+
+### Guidelines
+
+Before you submit your Pull Request (PR) consider the following guidelines:
+
+- Search GitHub for an open or closed PR that relates to your submission. You don't want to duplicate existing efforts.
+- Be sure that an existing issue describes the problem you're fixing, or documents the design for the feature you'd like to add. Discussing the design upfront helps to ensure that we're ready to accept your work. Make sure that this issue is referenced in your PR– this can be done by including `Closes: #[issue number]` in the PR's description.
+- Please sign our Contributor License Agreement (CLA) before sending PRs. We cannot accept code without a signed CLA. Make sure you author all contributed Git commits with email address associated with your CLA signature.
+
+### Style
+
+It is recommended that Pull Request titles follow the following format:
```

Comment:
We could add a note that we're using the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) format.